temporaltest/server_test.go

```diff
@@ -328,6 +340,9 @@ func TestSearchAttributeRegistration(t *testing.T) {
 	if err := wfr.Get(ctx, nil); err != nil {
 		t.Fatal(err)
 	}
+	// Wait a bit longer for the workflow to be indexed. This sleep usually isn't necessary,
+	// but helps avoid test flakiness.
+	time.Sleep(10 * time.Millisecond)
```

Comment:
The usual way we handle visibility lag in integration tests is to use `Eventually` from `github.com/stretchr/testify/assert` (or `require`) to just retry for a few seconds until it succeeds